test/AutoDiff/retroactive_derivatives.swift

```diff
@@ -0,0 +1,63 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+var RetroactiveDerivativeTests = TestSuite("RetroactiveDerivative")
```

Comment:
* Retroactive derivatives will be the only way to register derivatives in the future. Thus this test file name and test suite name should ideally be general like "derivative_registration.swift".
* There's a `@differentiating` test in `custom_derivatives.swift`. Do you want to move it here? We could also merge all derivative registration tests into `custom_derivatives.swift`.

Reply:
I moved all `@differentiating` tests to `test/AutoDiff/derivative_registration.swift`.

I chose not to move tests into `custom_derivatives.swift` because that file also contains tests for other APIs (`withGradient`, `withRecomputationInPullbacks`, etc).